filestore.go

```diff
@@ -0,0 +1,180 @@
+//go:build experimental
+// +build experimental
+
+// This file declares Go bindings for experimental features in TileDB.
+// Experimental APIs to do not fall under the API compatibility guarantees and
+// might change between TileDB versions
+
+package tiledb
+
+/*
+   	#cgo LDFLAGS: -ltiledb
+   	#cgo linux LDFLAGS: -ldl
+	#include <tiledb/tiledb_experimental.h>
+	#include <tiledb/tiledb_serialization.h>
+	#include <stdlib.h>
+*/
+import "C"
+import (
+	"errors"
+	"fmt"
+	"io"
+	"runtime"
+	"unsafe"
+)
+
+// FileStoreReader is an io.Reader for filestore files
+type FileStoreReader struct {
+	tdbCtx    *Context // the tiledb context for all read operations
+	arrayURI  string   // the uri of the array
+	arraySize int64    // the size of the array as returned by GetSize
+	bytesRead int64    // the total bytes read so far. Used as an offset for read operations
+}
+
+// NewFileStoreReader returns a FileStoreReader for the array
+// ArrayURI is the uri of an existing array with the filestore schema (see SchemaCreate)
+func NewFileStoreReader(tdbCtx *Context, arrayURI string) (*FileStoreReader, error) {
+	arraySize, err := GetSize(tdbCtx, arrayURI)
```

Comment:
Agreed, this will be further true for writes.

Reply:
For a reader, i like the check here. It is the equivalent of `os.Open` An error `array does not exist` is not inappropriate here. But in a subsequent `read` it will be.